parser-typechecker/src/Unison/Typechecker/Context.hs

```diff
@@ -264,6 +266,49 @@ data InfoNote v loc
   | TopLevelComponent [(v, Type.Type v loc, RedundantTypeAnnotation)]
   deriving (Show)
 
+topLevelComponent :: Var v => [(v, Type.Type v loc, RedundantTypeAnnotation)] -> InfoNote v loc
+topLevelComponent = TopLevelComponent . fmap (over _2 removeSyntheticTypeVars)
```

Comment:
This function is used when emitting type signatures for top-level definitions, and it does the `removeSyntheticTypeVars` pass.